guide/src/advanced/troubleshooting/cross_comp_config.md

```diff
@@ -0,0 +1,167 @@
+
+## Maximum Tx Size
+When Hermes relays packets or handshake messages, it will build multi-message Tx-es with up to `max_num_msgs` number of messages or up to a Tx size of `max_tx_size` bytes. The full node accepts only Tx-es with size up to the `max_tx_bytes` parameter in its `config.toml`. If a Tx is too large, the full node rejects them with the following error:
+
+```
+2022-10-27T13:59:43.650251Z ERROR ThreadId(18) send_messages_and_wait_commit{chain=ibc-0 tracking_id=ft-transfer}:send_tx_with_account_sequence_retry{chain=ibc-0 account.sequence=159}: gas estimation failed or encountered another unrecoverable error error=RPC error to endpoint http://127.0.0.1:26657/: response error: Internal error: Tx too large. Max size is 500, but got 615 (code: -32603)
+ERROR transfer error: failed while submitting the Transfer message to chain ibc-0: RPC error to endpoint http://127.0.0.1:26657/: response error: Internal error: Tx too large. Max size is 500, but got 615 (code: -32603)
+```
+### Fix
+Ensure that the `max_tx_size` parameter configured for Hermes is smaller than the `max_tx_bytes` parameter configured for the full node.
+
+## Allowed Clients
+The SDK chain genesis file contains a list of allowed clients that can be created on a chain. If the chain is configured with `genesis.app_state.ibc.client_genesis.allowed_clients` that doesn't include `07-tendermint`, then the chain will not allow tendermint IBC clients to be created and Hermes cannot open an IBC connections with this chain. When creating a client Hermes will get the following error:
+
+```
+2022-10-27T14:48:04.632320Z ERROR ThreadId(35) send_messages_and_wait_commit{chain=ibc-0 tracking_id=create client}:send_tx_with_account_sequence_retry{chain=ibc-0 account.sequence=0}:estimate_gas: failed to simulate tx. propagating error to caller: gRPC call failed with status: status: Unknown, message: "failed to execute message; message index: 0: client state type 07-tendermint is not registered in the allowlist: invalid client type [cosmos/ibc-go/v3@v3.0.0/modules/core/02-client/keeper/client.go:22] With gas wanted: '0' and gas used: '64671' ", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "x-cosmos-block-height": "9"} }
+```
+### Fix
+Ensure that the `genesis.app_state.ibc.client_genesis.allowed_clients` parameter in the chain genesis file includes `07-tendermint`.
+
+## Historical Entries
+Presence of recent consensus states on a chain is required when processing some connection handshake messages (`MsgConnectionOpenTry` and `MsgConnectionOpenAck`) that require proof verification that the counterparty chain has a valid client for this chain.
+If the chain is configured with `genesis.app_state.staking.params.historical_entries = 0`, then the chain will not store historical consensus state information and Hermes cannot open IBC connections with this chain. It will report an error message will look like this:
```

Comment:
```suggestion
If the chain is configured with `genesis.app_state.staking.params.historical_entries = 0`, then the chain will not store historical consensus state information and Hermes cannot open IBC connections with this chain. It will report an error message which looks like this:
```